tmt/base.py

```diff
@@ -4364,6 +4368,7 @@ def __init__(
         self.unique_id = str(time.time()).split('.')[0]
 
         self.policies = policies or []
+        self.recipe_builder = RecipeBuilder(logger) if self.opt('recipe') else None
```

Comment:
I am guessing the circular dependency comes here? I think it's easier to just `import` inside this function, or simply call it via `tmt.recipe.RecipeBuilder` (and you can type-hint it with that as well)